sql/src/test/java/org/apache/druid/sql/calcite/DrillWindowQueryTest.java

```diff
@@ -325,16 +326,20 @@ private static List<Object[]> parseResults(RowSignature rs, List<String[]> resul
         } else {
           switch (type.getType()) {
             case STRING:
```

Comment:
the else part should call parseElement function with 
`parseElement(val, type)`

ParseElement would look like 
```java
{
switch (elementType) {
      case STRING:
        return element;
      case LONG:
        return parseLongValue(element);
      case DOUBLE:
        return Numbers.parseDoubleObject(element);
       case Array:                 
              String[] elements = val.substring(1, val.length() - 1).split(","); // might need to adjust this.
              List<String> arrayElements = new ArrayList<>();
              for (String element : elements) {
                arrayElements.add(parseElement(element.trim(), type.getElementType().getType()).toString());
              }
            return "[" + String.join(",", arrayElements) + "]";
            default:
              throw new RuntimeException("unimplemented type: " + type.getType());
}
```
Then you can also handle nested arrays also if that is possible no ?

Reply:
@cryptoe Makes sense, have moved the `case ARRAY` logic to `parseElement()` method itself as suggested.

I think it would require some more work to make it work for nested arrays though, as that would need a `case COMPLEX` logic. But I think we can do that in future when/if we need to, as it's outside the intended scope of this PR.

Reply:
Should the nullHandling be also pushed inside the `parseElement` call ?

Reply:
@cryptoe Have made the change.